frontend/test/pytest/test_static_arguments.py

```diff
@@ -221,6 +222,55 @@ def wrapper(x, c):
         captured = capsys.readouterr()
         assert captured.out.strip() == "Inside QNode: 0.5"
 
+    def test_static_argnames(self):
+        # pylint: disable=inconsistent-return-statements
+        """Test static arguments specified by names"""
+
+        @qjit(static_argnames="y")
+        def f(x, y):
+            if y < 10:
+                return x + y
+
+        assert f(1, 2) == 3
+
+        @qjit(static_argnames="x")
+        def g(x, y):
+            if y < 10:
+                return x + y
+
+        with pytest.raises(
+            TracerBoolConversionError, match="Attempted boolean conversion of traced"
+        ):
+            g(1, 2)
+
+        @qjit(static_argnames=("x", "y"))
+        def h(x, y):
+            if x < 10 and y < 10:
+                return x + y
+
+        assert h(1, 2) == 3
+
+        @qjit(static_argnames=("x"), static_argnums=[1])
+        def p(x, y):
+            if x < 10 and y < 10:
+                return x + y
+
+        assert p(1, 2) == 3
+
+        @qjit(static_argnames=("y"), static_argnums=[0])
+        def q(x, y):
+            if x < 10 and y < 10:
+                return x + y
+
+        assert q(1, 2) == 3
+
+        @qjit(static_argnames=("y"), static_argnums=[1])
+        def r(x, y):
+            if y < 10:
+                return x + y
+
+        assert r(1, 2) == 3
```

Comment:
@paul0403 have you tested that it works with different combinations of defining and calling functions? E.g.,

- defining functions that use `*` and `/` in the signature
- defining functions that use default values in the signature
- calling with a mixture of keyword and positional values

Reply:
Through testing this I realized currently `static_argnums` do not support calls with default arguments (this is on main):
```python
@qjit(static_argnums=[1])
def f(y, x=9):
    if x < 10:
        return x + y
    return 42000

res = f(20)
print(res)
```

```
catalyst.utils.exceptions.CompileError: argnum 1 is beyond the valid range of [0, 1).
```

Note that jax works:
```python
@partial(jax.jit, static_argnums=[1])
def f(y, x=9):
    if x < 10:
        return x + y
    return 42000


res = f(20)
print(res)
```

```
29
```

I don't think this is an issue with `static_argname`. I propose we open a separate issue to fix it.

Reply:
Nice, thanks @paul0403! My intuition was that this is worth testing as edge cases/bugs like this might appear :)

If this affects `static_argnum`, and would be resolved here once `static_argnum` is fixed, I am okay opening a separate issue to fix it.

Reply:
So I think none of the more complicated patterns work, but not because of `static_argname`, but because they never worked with `static_argnum` to begin with.

One particular case was handed out as an assessment: #1163 . In the issue I documented the failure. The apparent failure is in a verification, but I am not sure whether this means (a) the underlying mechanism is buggy and the verification failed as a by-product, or (b) the underlying mechanism is ok, but the verification itself is overly strict.

Reply:
@josh146 I suggest we finish the simple base case of `static_argnames` for release, since the other failed cases were already there with `static_argnums` as well.

Reply:
that works for me!

Reply:
@paul0403 if you haven't already, would you be able to open an issue detailing the more advanced cases that don't work with `static_argnums`? https://github.com/PennyLaneAI/catalyst/pull/1158#discussion_r1781382066

Reply:
> @paul0403 if you haven't already, would you be able to open an issue detailing the more advanced cases that don't work with `static_argnums`? [#1158 (comment)](https://github.com/PennyLaneAI/catalyst/pull/1158#discussion_r1781382066)

Already did, and actually we sent this out as an assessment! #1163

Reply:
(As far as I can tell, the default value issue seems like the root cause of all other patterns' failures.)